eventlet/greenio.py

```diff
@@ -1,693 +1,14 @@
-import errno
-import os
-from socket import socket as _original_socket
-import socket
-import sys
-import time
-import warnings
+from eventlet.support import six
 
-from eventlet.support import get_errno, six
-from eventlet.hubs import trampoline, notify_close, notify_opened, IOClosed
+from eventlet._basegreenio import _GLOBAL_DEFAULT_TIMEOUT  # noqa
+from eventlet._basegreenio import *  # noqa
 
-__all__ = ['GreenSocket', 'GreenPipe', 'shutdown_safe']
-
-BUFFER_SIZE = 4096
-CONNECT_ERR = set((errno.EINPROGRESS, errno.EALREADY, errno.EWOULDBLOCK))
-CONNECT_SUCCESS = set((0, errno.EISCONN))
-if sys.platform[:3] == "win":
-    CONNECT_ERR.add(errno.WSAEINVAL)   # Bug 67
-
-if six.PY3:
-    from io import IOBase as file
-    _fileobject = socket.SocketIO
-elif six.PY2:
-    _fileobject = socket._fileobject
-
-
-def socket_connect(descriptor, address):
-    """
-    Attempts to connect to the address, returns the descriptor if it succeeds,
-    returns None if it needs to trampoline, and raises any exceptions.
-    """
-    err = descriptor.connect_ex(address)
-    if err in CONNECT_ERR:
-        return None
-    if err not in CONNECT_SUCCESS:
-        raise socket.error(err, errno.errorcode[err])
-    return descriptor
-
-
-def socket_checkerr(descriptor):
-    err = descriptor.getsockopt(socket.SOL_SOCKET, socket.SO_ERROR)
-    if err not in CONNECT_SUCCESS:
-        raise socket.error(err, errno.errorcode[err])
-
-
-def socket_accept(descriptor):
-    """
-    Attempts to accept() on the descriptor, returns a client,address tuple
-    if it succeeds; returns None if it needs to trampoline, and raises
-    any exceptions.
-    """
-    try:
-        return descriptor.accept()
-    except socket.error as e:
-        if get_errno(e) == errno.EWOULDBLOCK:
-            return None
-        raise
-
-
-if sys.platform[:3] == "win":
-    # winsock sometimes throws ENOTCONN
-    SOCKET_BLOCKING = set((errno.EAGAIN, errno.EWOULDBLOCK,))
-    SOCKET_CLOSED = set((errno.ECONNRESET, errno.ENOTCONN, errno.ESHUTDOWN))
+if six.PY2:
+    from eventlet._py2greenio import _fileobject  # noqa
```

Comment:
Intuitively, `import *` should be enough, without `_fileobject`. Is there any reason why `_fileobject` is not included in `_pyXgreenio.__all__`?